team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							const moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGenBreeding && (dexEntry.tier.startsWith('Bank') || moveGen !== '7' || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
Sorry for making you go through all these changes, by the way. It's just that code readability is really important for functions like these; refactoring this function last time was a nightmare because people pushed unreadable code.